docs/docs/adding-search-with-algolia.md

```diff
@@ -2,593 +2,593 @@
+
+It is also best practice not to check in the `.env` file for this reason. Consider creating an `.env.example` without the values to git instead. This way, if someone else sets up the project, they know what configuration they need to supply but don't have access to your private values.
+
+Next, modify `gatsby-config.js` to read the configuration and add the `gatsby-plugin-algolia` plugin.
 
+Add the following line at the top of `gatsby-config.js` to read the configuration from `.env`:
+
+```js:title=gatsby-config.js
 require("dotenv").config()
+```
 
-module.exports = {
-  siteMetadata: {
-    title: `Gatsby+Algolia`,
-    description: `How to setup Algolia search in Gatsby`,
-    author: `<your name>`,
-  },
+Then add the configuration for `gatsby-plugin-algolia` to the list of `plugin`s in the `gatsby-config.js`. `dotenv` has made the configuration values available as keys in `process.env`.
+
+```js:title=gatsby-config.js
   plugins: [
+    ... // your existing plugins here
     {
       resolve: `gatsby-plugin-algolia`,
       options: {
         appId: process.env.GATSBY_ALGOLIA_APP_ID,
         apiKey: process.env.ALGOLIA_ADMIN_KEY,
-        queries,
-        chunkSize: 10000, // default: 1000
+        queries: require("./src/utils/algolia-queries")
       },
-    },
-    `gatsby-plugin-styled-components`,
+    }
   ],
 }
 ```
 
-Notice that you're loading `queries` from a file at `./src/utils/algolia.js` (you can of course put it wherever you like) and your Algolia ID and API key from `.env` so add those files.
+### Query the pages for indexing
 
-For this, you will need to navigate to [the 'API Keys' section of your Algolia profile](https://www.algolia.com/api-keys). If you already have an account, you will find your API keys here. If not, you will need to sign up for one and then navigate to this link. It should look something like this screenshot, only with actual numbers instead of redacted ones:
+You still need to supply a `queries` configuration. Queries tell the Algolia plugin what data is to be indexed. They perform GraphQL queries for the relevant pages and convert the response into a set of Algolia records. These contain key/value pairs with the data to be indexed.
 
-![Algolia API key screenshot](./images/algolia-api-keys.png)
+The configuration could have been entered straight into the `gatsby-config.js`, but the configuration above loads it from a new file `src/utils/algolia-queries.js` to avoid clutter. Create this page in your project:
 
-Once you have your App ID, Search-Only API Key, and Admin API Key, place the following code into your `.env` file, replacing the placeholder keys with your keys:
-
-```text:title=.env
-GATSBY_ALGOLIA_APP_ID=KA4OJA9KAS
-GATSBY_ALGOLIA_SEARCH_KEY=lkjas987ef923ohli9asj213k12n59ad
-ALGOLIA_ADMIN_KEY=lksa09sadkj1230asd09dfvj12309ajl
-```
+```js:title=src/utils/algolia-queries.js
+const escapeStringRegexp = require("escape-string-regexp")
```

Comment:
Doesn't this require an npm install as well?

Reply:
Gatsby already includes this library so it will be available to the project code. You can argue that it should nevertheless explicitly declare that it uses it directly, but I don't think it's worth complicating the tutorial for it. 

Since we figured out it will generally work without escaping, too, we could consider dropping the escaping completely if you think it complicates the story.